x-pack/apm-server/sampling/eventstorage/storage_manager.go

```diff
@@ -26,25 +25,46 @@ const (
 	// subscriberPositionFile holds the file name used for persisting
 	// the subscriber position across server restarts.
 	subscriberPositionFile = "subscriber_position.json"
-)
 
-var (
-	errDropAndRecreateInProgress = errors.New("db drop and recreate in progress")
+	// partitionsPerTTL holds the number of partitions that events in 1 TTL should be stored over.
+	// Increasing partitionsPerTTL increases read amplification, but decreases storage overhead,
+	// as TTL GC can be performed sooner.
+	//
+	// For example, partitionPerTTL=1 means we need to keep 2 partitions active,
+	// such that the last entry in the previous partition is also kept for a full TTL.
+	// This means storage requirement is 2 * TTL, and it needs to read 2 keys per trace ID read.
+	// If partitionPerTTL=2, storage requirement is 1.5 * TTL at the expense of 3 reads per trace ID read.
+	partitionsPerTTL = 1
```

Comment:
[to reviewer] e.g. pertitionsPerTTL=3 creates 4 active partitions, and 1 can be cleared out per rotation, i.e. 33% of entries can be cleared at `TTL/3` interval. This yields the most stable disk usage in @kruskall 's experiments, but I'm slightly worried about the read amplification, which is not well reflected in the experiment as it wasn't tested with remote sampling decisions. While it is possible to microbenchmark this, but I doubt how much it reflects the reality because of pebble block cache.